src/Nethermind/Nethermind.Blockchain/Find/LogFinder.cs

```diff
@@ -63,18 +63,29 @@ public class LogFinder : ILogFinder
         public IEnumerable<FilterLog> FindLogs(LogFilter filter, CancellationToken cancellationToken = default)
         {
             BlockHeader FindHeader(BlockParameter blockParameter, string name, bool headLimit) => 
-                _blockFinder.FindHeader(blockParameter, headLimit) ?? throw new ArgumentException("Block not found.", name);
+                _blockFinder.FindHeader(blockParameter, headLimit) ?? throw new ResourceNotFoundException($"Block not found: {name}");
 
             cancellationToken.ThrowIfCancellationRequested();
-            var toBlock = FindHeader(filter.ToBlock, nameof(filter.ToBlock), true);
+            var toBlock = FindHeader(filter.ToBlock, nameof(filter.ToBlock), false);
```

Comment:
Now its possible for this to return null too. Shouldn't we throw here too?

Reply:
It will throw in line 66.